### PR TITLE
catch before running success callback to avoid catching errors that occur in success callback

### PR DESCRIPTION
**Pull request by gravitypersists**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,4 +27,5 @@ export default function fancyFetch(options) {
       error({ networkFailure: true })
     }
   })
+  .then(json => success(json))
```
Comment (tobyhs) [inline index.js]: I still don't know JavaScript well, but doesn't having the `catch` right before this `then` cause the `then` to run even if there was an HTTP error? (I'm looking at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch#Examples and seeing `'after a catch the chain is restored'`)

**Comment by yiziz** [comment]
What does changing the order do?  Also is there a way to prevent catching syntax errors or other simple mistakes? would really help with debugging

**Comment by yiziz** [comment]
:+1:

**Comment by tobyhs** [comment]
@yiziz Doesn't test code catch syntax errors and simple mistakes?

**Comment by yiziz** [comment]
@tobyhs I think it will swallow all errors when we use `fancyFetch` with `success` callback function that contains a syntax error.

File: index.js (hunk 1 above)
Comment (tobyhs) [inline index.js]: A followup example:
```
> let promise = new Promise(() => { throw new Error('err') })
undefined
> promise.catch((e) => { console.log('eee') }).then(() => { console.log('succ') })
Promise { <pending> }
> eee
succ
```

**Comment by gravitypersists** [comment]
Toby's right, this isn't the fix we're looking for 

```
var p1 = new Promise(function(resolve, reject) {
  reject('Network failure');
});

p1.then(function(value) {
  return new Promise(function(resolve, reject) {
    reject('Response error')
  })
}).catch(function(e) {
  console.log(e);
}).then(function(){
  console.log('success callback is fired anyways');
});
```

**Pull request closed without merging**